src/core/thread/mle.cpp

```diff
@@ -2099,6 +2099,10 @@ ThreadError Mle::HandleChildIdResponse(const Message &aMessage, const Ip6::Messa
             mNetif.GetActiveDataset().Set(activeTimestamp, aMessage, offset + sizeof(tlv), tlv.GetLength());
         }
     }
+    else
```

Comment:
This is done as a part of `ActiveDataset::Set()`.  If the local active timestamp is newer, `ActiveDataset::Set()` will trigger sending a MGMT_ACTIVE_SET.req message to the leader.